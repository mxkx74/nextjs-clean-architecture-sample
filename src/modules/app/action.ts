import type { ActionTypes } from '../../types/action';

const types = {
  INIT_APP: 'INIT_APP',
} as const;

const initApp = () => {
  return {
    type: types.INIT_APP,
  };
};

export const actionCreators = {
  initApp,
};

export type AppAction = ActionTypes<typeof actionCreators>;
