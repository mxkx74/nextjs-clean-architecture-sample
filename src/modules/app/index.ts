export { actionCreators } from './action';
export { reducer } from './reducer';
export type { AppAction } from './action';
export type { AppState } from './reducer';
