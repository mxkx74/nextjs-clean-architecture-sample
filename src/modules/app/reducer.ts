import produce from 'immer';
import type { AppAction } from './action';

export type AppState = {
  initialize: boolean;
};

export const initialState: AppState = {
  initialize: false,
};

export const reducer = produce((draft: AppState, action: AppAction) => {
  switch (action.type) {
    case 'INIT_APP':
      draft.initialize = true;
      break;
  }
}, initialState);
