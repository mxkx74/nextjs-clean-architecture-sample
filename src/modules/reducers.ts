import { AppState, reducer as app } from './app/index';

export type StoreState = Readonly<{
  app: AppState;
}>;

export const reducers = {
  app,
};
