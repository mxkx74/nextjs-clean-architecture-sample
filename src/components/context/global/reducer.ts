import { GlobalActions } from './action';

export type GlobalState = {
  value?: string;
};

export const initialState: GlobalState = {};

export const reducer = (state: GlobalState, action: GlobalActions) => {
  switch (action.type) {
    case 'SET_VALUE':
      return {
        ...state,
        value: action.payload.value,
      };
    default:
      return state;
  }
};
