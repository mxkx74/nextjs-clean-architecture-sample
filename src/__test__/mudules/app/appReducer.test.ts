import snapshotDiff from 'snapshot-diff';
import { actionCreators, reducer } from '../../../modules/app/';
import { initialState } from '../../../modules/app/reducer';

describe('appのテスト', () => {
  const action = actionCreators.initApp();
  const data = reducer(initialState, action);

  it('起動時にinitializeがtrueになる', () => {
    expect(data.initialize).toBeTruthy();
  });

  it('スナップショット', () => {
    expect(snapshotDiff(initialState, data)).toMatchSnapshot();
  });
});
