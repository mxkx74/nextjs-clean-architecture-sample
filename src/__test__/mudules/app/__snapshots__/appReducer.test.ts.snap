// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appのテスト スナップショット 1`] = `
"Snapshot Diff:
- First value
+ Second value

  Object {
-   \\"initialize\\": false,
+   \\"initialize\\": true,
  }"
`;
